ttl/ttlworker/task_manager.go

```diff
@@ -328,12 +348,26 @@ func (m *taskManager) lockScanTask(se session.Session, task *cache.TTLTask, now
 	err := se.RunInTxn(ctx, func() error {
 		var err error
 
+		// The `for update` here ensures two things:
+		// 1. The instance which will update this row has committed, and we can get the newest information.
+		// 2. There is only one successful transaction concurrently.
+		//    If it's a `for update nowait`, we cannot avoid the situation that multiple transactions fetched the tasks
+		//    at the same time, and the task limitation doesn't work.
```

Comment:
I think these comments should be moved into `syncTaskFromTable`

Reply:
tidb does not support gapped lock, is it possible to avoid the task count to exceed the limit?

Reply:
> I think these comments should be moved into `syncTaskFromTable`

There comments are here to describe how we make the limitation work, and the limitation is checked and considered in `lockScanTask`, so I kept these comments in the `lockScanTask`.

> tidb does not support gapped lock, is it possible to avoid the task count to exceed the limit?

How is it connected with gapped lock? This is only a point get (with `for update`) :thinking:

Reply:
Discussed with lcwangchao. Resolved.